File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/AllTasksListAsyncProvider.java
package de.szalkowski.activitylauncher;

import android.content.Context;

public class AllTasksListAsyncProvider extends AsyncProvider<AllTasksListAdapter> {
    private AllTasksListAdapter adapter;

    AllTasksListAsyncProvider(
            Context context,
            de.szalkowski.activitylauncher.AsyncProvider.Listener<AllTasksListAdapter> listener) {
        super(context, listener, true);
        this.adapter = new AllTasksListAdapter(context);
    }

    @Override
    protected AllTasksListAdapter run(Updater updater) {
        this.adapter.resolve(updater);
        return this.adapter;
    }
}

File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/AllTasksListAsyncProvider.kt
package de.szalkowski.activitylauncher

import android.content.Context

class AllTasksListAsyncProvider internal constructor(
        context: Context,
        listener: AsyncProvider.Listener<AllTasksListAdapter>) : AsyncProvider<AllTasksListAdapter>(context, listener, true) {
    private val adapter: AllTasksListAdapter

    init {
        this.adapter = AllTasksListAdapter(context)
    }

    override fun run(updater: AsyncProvider<AllTasksListAdapter>.Updater): AllTasksListAdapter {
        this.adapter.resolve(updater)
        return this.adapter
    }
}
